src/transformStateWithClones.js

```diff
@@ -7,7 +7,50 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  let lastObject;
+  let nextObject;
+  const resultArray = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i];
+
+    if (i === 0) {
+      lastObject = { ...stateCopy };
+    } else {
+      lastObject = { ...resultArray[i - 1] };
+    }
+
+    switch (true) {
```

Comment:
Use 'switch (action.type)' instead of 'switch (true)' for more readability and better performance.

Reply:
fixed